spec/features/cases/foi/case_assignment_responding_spec.rb

```diff
@@ -123,4 +123,14 @@
     expect(page).to_not have_content('This case has already been rejected.')
   end
 
+  scenario 'kilo clicks on a link to an assignment that has been deleted' do
+    assignment_id = assignment.id
+    assignment.accept responder
+    assigned_case.update_attribute(:deleted, true)
+
+    visit edit_case_assignment_path assigned_case, assignment_id
+
+    expect(page).to have_current_path(open_filter_path)
+    expect(page).to have_content("Case has been deleted.")
```

Comment:
See patch diff - this test is not technically correct because the soft-delete gem will treat `Case::Base.find(id)` as a 'not found exception'. The test above works because `assigned_case` is still in memory/cache during the test and therefore the corresponding code in the controller is required to do the `Case::Base.unscoped.find(id)` rather than just `Case::Base.exists?(id)` or `Case::Base.find_by(id: id)`

See patch file for what the test/corresponding method look like

Reply:
Pinged you on Slack :)

Reply:
As discussed, IMO the test was correct but I've changed it to your suggested approach which also works fine here 👍